lib/src/layer/polyline_layer.dart

```diff
@@ -41,64 +64,185 @@ class Polyline {
+  /// (via [ValueNotifier.value]) in the detector's callbacks. It is also
+  /// possible to listen to the notifier directly.
+  ///
+  /// Note that a hover event is included as a hit event. Therefore for
+  /// performance reasons, it may be advantageous to check the new value's
+  /// equality against the previous value (excluding the [PolylineHit.point],
+  /// which will always change), and avoid doing any heavy work if they are the
+  /// same.
+  ///
+  /// See online documentation for more detailed usage instructions. See the
+  /// example project for an example implementation.
+  ///
+  /// Will notify with [PolylineHit]s when any [Polyline]s are hit, otherwise
+  /// will notify with `null`.
+  final PolylineHitNotifier? hitNotifier;
+
+  /// The minimum radius of the hittable area around each [Polyline] in logical
+  /// pixels
+  ///
+  /// The entire visible area is always hittable, but if the visible area is
+  /// smaller than this, then this will be the hittable area.
+  ///
+  /// Defaults to 10.
+  final double minimumHitbox;
 
   const PolylineLayer({
     super.key,
     required this.polylines,
-    this.polylineCulling = false,
+    this.hitNotifier,
+    this.minimumHitbox = 10,
+    // TODO: Remove once PR #1704 is merged
+    bool polylineCulling = true,
   });
 
   @override
   Widget build(BuildContext context) {
-    final map = MapCamera.of(context);
+    final camera = MapCamera.of(context);
 
     return MobileLayerTransformer(
       child: CustomPaint(
-        painter: PolylinePainter(
-          polylineCulling
-              ? polylines
-                  .where((p) => p.boundingBox.isOverlapping(map.visibleBounds))
-                  .toList()
-              : polylines,
-          map,
+        painter: _PolylinePainter(
+          polylines: polylines
+              .where((p) => p.boundingBox.isOverlapping(camera.visibleBounds))
+              .toList(),
+          camera: camera,
+          hitNotifier: hitNotifier,
+          minimumHitbox: minimumHitbox,
         ),
-        size: Size(map.size.x, map.size.y),
+        size: Size(camera.size.x, camera.size.y),
         isComplex: true,
       ),
     );
   }
 }
 
-class PolylinePainter extends CustomPainter {
+class _PolylinePainter extends CustomPainter {
   final List<Polyline> polylines;
-
-  final MapCamera map;
+  final MapCamera camera;
   final LatLngBounds bounds;
+  final PolylineHitNotifier? hitNotifier;
+  final double minimumHitbox;
 
-  PolylinePainter(this.polylines, this.map) : bounds = map.visibleBounds;
+  _PolylinePainter({
+    required this.polylines,
+    required this.camera,
+    required this.hitNotifier,
+    required this.minimumHitbox,
+  }) : bounds = camera.visibleBounds;
 
-  int get hash => _hash ??= Object.hashAll(polylines);
+  // Avoids reallocation on every `hitTest`, is cleared every time
+  final hits = List<Polyline>.empty(growable: true);
```

Comment:
Probably better to use `///` here

Reply:
```suggestion
  /// Avoids reallocation on every `hitTest`, is cleared every time
  final hits = List<Polyline>.empty(growable: true);
```

Reply:
Its internal anyway so it won't make a difference. The comment also only explains why it exists, not what it does, so probably not suitable for a doc string.